Remotion/Core/Core/Collections/DictionaryExtensions.cs

```diff
@@ -27,33 +28,44 @@ namespace Remotion.Collections
   /// </summary>
   public static class DictionaryExtensions
   {
+    [return: MaybeNull]
     public static TValue GetValueOrDefault<TKey, TValue> (this IDictionary<TKey, TValue> dictionary, TKey key)
+        where TKey : notnull
     {
+      //TODO: add argument null checks for key values for all methods
       ArgumentUtility.CheckNotNull ("dictionary", dictionary);
       // Implementations of IDictionary<TKey, TValue> are free to allow null keys.
 
-      return GetValueOrDefault (dictionary, key, default (TValue));
+      return GetValueOrDefault (dictionary, key, default (TValue)!);
     }
 
+    [return: MaybeNull]
     public static TValue GetValueOrDefault<TKey, TValue> (this IReadOnlyDictionary<TKey, TValue> dictionary, TKey key)
+        where TKey : notnull
     {
       ArgumentUtility.CheckNotNull ("dictionary", dictionary);
       // Implementations of IDictionary<TKey, TValue> are free to allow null keys.

```

Comment:
remove those comments, add ticket-no to the todo, probably add todo to every line where we want it.

Reply:
same ticket as improvements or null bugs?

Reply:
yeah